catroid/src/main/java/org/catrobat/catroid/CatroidApplication.java

```diff
@@ -49,6 +58,162 @@ public void onCreate() {
 		Log.d(TAG, "CatroidApplication onCreate");
 		settings = new ApplicationSettings(this);
 		CatroidApplication.context = getApplicationContext();
+		defaultSystemLanguage = Locale.getDefault().getLanguage();
+		// open the App in the last chosen language
+		languageSharedPreferences = getSharedPreferences("For_language", getApplicationContext().MODE_PRIVATE);
+		String langTag = languageSharedPreferences.getString("Nur", "");
+		switch (langTag) {
```

Comment:
To avoid such huge switch statements with a lot of the same code over and over, you could store all possible languages in a static final array or maybe as a resource and check if the array contains a string equal to langTag - if it does: "Multilingual.setContextLocale(getApplicationContext(), langTag);"

If it doesn't - check for "".